specs/typography.md

```diff
@@ -0,0 +1,177 @@
+- [Material Design](https://material.io/design/typography/the-type-system.html)
+- [Bootstrap](https://getbootstrap.com/docs/4.4/content/typography)
+- [Ant Design](https://ant.design/components/typography)
+- [Semantic UI](https://semantic-ui.com/elements/header.html)
+- [Fluent UI](https://fluentsite.z22.web.core.windows.net/components/header/definition)
+
+Most other component frameworks either maintain the base font size or set it smaller to 14px. They also typically maintain heading element sizes based on `em` or `rem`, though some use individual specific values. Typically heading sizes are close to the browser default, though Material Design goes a lot larger at the top.
+
+---
+
+## Design
+
+The type styles published are consistently named so they are easy to identify and consume. A clear separation is made between the semantic naming and the raw type ramp.
+
+Default styles are applied to standard elements such that headings and other base typography pick up the adaptive styling without needing to add class names.
+
+(**Editorial**: can we do this, or do we need to build this support?)
+
+### ~~API~~
+
+*~~The key elements of the component's public API surface~~*
+
+### Adaptive Styling Properties (**Editorial**: Added this section)
+
+- `type-ramp-base-size` = "14px"  
+The base size for primary body text. Other type ramp sizes may eventually be determined from this starting value.
+
+### Anatomy and Appearance
+
+The type ramp is constructed as css variables that go plus and minus from the base. This allows a consumer to extend the ramp if necessary.
+
+This ramp is currently based on fixed values according to the current visual design, but could be updated from an algorithm if desired.
+
+- `fast-type-ramp-base-size` = `type-ramp-base-size`
+- `fast-type-ramp-plus-1-size` = "16px"
+- `fast-type-ramp-plus-2-size` = "20px"
+- `fast-type-ramp-plus-3-size` = "28px"
+- `fast-type-ramp-plus-4-size` = "34px"
+- `fast-type-ramp-plus-5-size` = "46px"
+- `fast-type-ramp-plus-6-size` = "60px"
+- `fast-type-ramp-minus-1-size` = "12px"
+- `fast-type-ramp-minus-2-size` = "10px"
+
+The semantically named styles that are intended to be consumed at the component level:
+
+- `fast-display-type-size` = `fast-type-ramp-plus-6-size`
+- `fast-header-type-size` = `fast-type-ramp-plus-5-size`
+- `fast-subheader-type-size` = `fast-type-ramp-plus-4-size`
+- `fast-page-title-type-size` = `fast-type-ramp-plus-3-size`
+- `fast-section-title-type-size` = `fast-type-ramp-plus-2-size`
+- `fast-paragraph-larger-type-size` = `fast-type-ramp-plus-1-size`
+- `fast-paragraph-type-size` = `fast-type-ramp-base-size`
+- `fast-paragraph-smaller-type-size` = `fast-type-ramp-minus-1-size`
+- `fast-label-type-size` = `fast-type-ramp-base-size`
+- `fast-button-type-size` = `fast-type-ramp-base-size`
+- `fast-caption-type-size` = `fast-type-ramp-minus-1-size`
+- `fast-caption-smaller-type-size` = `fast-type-ramp-minus-2-size`
```

Comment:
Can you provide comments as to when these each would be used? Is there a pattern that is being proposed for how this might extend in the future if need be?

Reply:
I'm also curious how this scales beyond our own internal principles? At first glance, this seems very opinionated compared to the current implementation of typography.